app/src/main/java/com/github/doomsdayrs/apps/shosetsu/backend/TarArchive.kt

```diff
@@ -0,0 +1,83 @@
+package com.github.doomsdayrs.apps.shosetsu.backend
+
+import org.kamranzafar.jtar.*
+import java.io.File
+import java.io.InputStream
+import java.io.RandomAccessFile
+
+class TarArchive(val file: File)
```

Comment:
Instead of constructor taking in a file it might be better to inherit File and then override a bunch of its methods.